kolibri/core/auth/test/test_morango_integration.py

```diff
@@ -1053,3 +795,94 @@ def assert_existence(self, server, kind, assignment_id, should_exist=True):
             ), "Assignment {assignment_id} should exist on server {server}!".format(
                 assignment_id=assignment_id, server=server
             )
+
+
+@unittest.skipIf(
+    not os.environ.get("INTEGRATION_TEST"),
+    "This test will only be run during integration testing.",
+)
+class SingleUserSyncRegressionsTestCase(TestCase):
+    @multiple_kolibri_servers(2)
+    def test_facility_user_conflict_syncing_from_tablet(self, servers):
+        self._test_facility_user_conflict(servers, True)
+
+    @multiple_kolibri_servers(2)
+    def test_facility_user_conflict_syncing_from_laptop(self, servers):
+        self._test_facility_user_conflict(servers, False)
+
+    def _test_facility_user_conflict(self, servers, tablet_is_client):
+        """
+        This is a regression test to handle the case of a FacilityUser being changed
+        on a SoUD as a backend side effect, e.g. by Django updating the `last_login`
+        field, leading to a merge conflict when the user is updated on another device.
+        """
+
+        laptop, tablet = servers
+
+        facility, learner, _ = laptop.generate_base_data()
+
+        tablet.sync(laptop, facility, user=learner)
+
+        laptop.update_model(FacilityUser, learner.id, full_name="NEW")
+
+        tablet.update_model(
+            FacilityUser, learner.id, last_login=datetime.datetime.now()
+        )
+
+        if tablet_is_client:
+            tablet.sync(laptop, facility, user=learner)
+        else:
+            laptop.sync(tablet, facility, user=learner)
+
+        assert (
+            FacilityUser.objects.using(tablet.db_alias).get(id=learner.id).full_name
+            == "NEW"
+        )
+
+    @multiple_kolibri_servers(3)
+    def test_kolibri_issue_8439(self, servers):
```

Comment:
This is the actual new test that was introduced here. It's short but effective at catching the issue.